src/Component/Provider/ArrayGridProvider.php

```diff
@@ -37,22 +44,20 @@ public function get(string $code): Grid
         }
 
         $gridConfiguration = $this->gridConfigurations[$code];
+        $parentGridCode = $gridConfiguration['extends'] ?? null;
+
+        if (null !== $parentGridCode) {
+            $parentGridConfiguration = $this->gridConfigurations[$gridConfiguration['extends']] ?? null;
 
-        if (isset($gridConfiguration['extends'], $this->gridConfigurations[$gridConfiguration['extends']])) {
-            $gridConfiguration = $this->extend($gridConfiguration, $this->gridConfigurations[$gridConfiguration['extends']]);
+            Assert::notNull($parentGridConfiguration, sprintf('Parent grid with code "%s" does not exists.', $gridConfiguration['extends']));
+            $gridConfiguration = $this->extend($gridConfiguration, $parentGridConfiguration);
         }
 
         return $this->converter->convert($code, $gridConfiguration);
     }
 
     private function extend(array $gridConfiguration, array $parentGridConfiguration): array
     {
-        unset($parentGridConfiguration['sorting']); // Do not inherit sorting.
-
-        $configuration = array_replace_recursive($parentGridConfiguration, $gridConfiguration) ?: [];
-
-        unset($configuration['extends']);
-
-        return $configuration;
+        return $this->gridConfigurationExtender->extends($gridConfiguration, $parentGridConfiguration);
```

Comment:
This method can be removed, it doesn't do anything.

Reply:
done